lib/reline.rb

```diff
@@ -400,9 +390,8 @@ def readline(prompt = '', add_hist = false)
         end
         case result
         when :matched
-          expanded = key_stroke.expand(buffer).map{ |expanded_c|
-            Reline::Key.new(expanded_c, expanded_c, false)
-          }
+          expanded, rest_bytes = key_stroke.expand(buffer)
+          rest_bytes.reverse_each { |c| io_gate.ungetc(c) }
           block.(expanded)
```

Comment:
Do we need a `if expanded` here as well?

Reply:
I changed `expanded, rest_bytes = key_stroke.expand(...)` to always return a pair of array.